R-package/R/xgb.DMatrix.R

```diff
@@ -81,6 +107,9 @@ xgb.DMatrix <- function(data, info = list(), missing = NA, silent = FALSE, nthre
   if (!is.null(cnames)) {
     setinfo(dmat, "feature_name", cnames)
   }
+  if (!is.null(ctypes)) {
+    setinfo(dmat, "feature_type", ctypes)
```

Comment:
Worth adding to the docs that these are automatically determined for data frames.

Reply:
I will add the `enable_categorical` parameter similar to the one in Python. Do you prefer `enable_categorical` or `enable.categorical` or something else?

Reply:
I think `enable_categorical` would be better but don't have any strong preference.